gb-branch/public/scripts/articleView.js

```diff
@@ -0,0 +1,144 @@
+        $('article.template').hide();
+      }
+      $('#author-filter').val('');
+    });
+  };
+
+  articleView.handleMainNav = () => {
+    $('nav').on('click', '.tab', function(e) {
+      e.preventDefault();
+      $('.tab-content').hide();
+      $(`#${$(this).data('content')}`).fadeIn();
+    });
+
+    $('nav .tab:first').click();
+  };
+
+  articleView.setTeasers = () => {
+    $('.article-body *:nth-of-type(n+2)').hide();
+    $('article').on('click', 'a.read-on', function(e) {
+      e.preventDefault();
+      if ($(this).text() === 'Read on →') {
+        $(this).parent().find('*').fadeIn();
+        $(this).html('Show Less &larr;');
+      } else {
+        $('body').animate({
+          scrollTop: ($(this).parent().offset().top)
+        },200);
+        $(this).html('Read on &rarr;');
+        $(this).parent().find('.article-body *:nth-of-type(n+2)').hide();
+      }
+    });
+  };
+
+  articleView.initNewArticlePage = () => {
+    $('.tab-content').show();
+    $('#export-field').hide();
+    $('#article-json').on('focus', function(){
+      this.select();
+    });
+
+    $('#new-form').on('change', 'input, textarea', articleView.create);
+    $('#new-form').on('submit', articleView.submit);
+  };
+
+  articleView.create = () => {
+    var article;
+    $('#articles').empty();
+
+    article = new Article({
+      title: $('#article-title').val(),
+      author: $('#article-author').val(),
+      authorUrl: $('#article-author-url').val(),
+      category: $('#article-category').val(),
+      body: $('#article-body').val(),
+      publishedOn: new Date().toISOString()
+    });
+
+    $('#articles').append(article.toHtml());
+    $('pre code').each((i, block) => hljs.highlightBlock(block));
+  };
+
+  articleView.submit = event => {
+    event.preventDefault();
+    let article = new Article({
+      title: $('#article-title').val(),
+      author: $('#article-author').val(),
+      authorUrl: $('#article-author-url').val(),
+      category: $('#article-category').val(),
+      body: $('#article-body').val(),
+      publishedOn: new Date().toISOString()
+    });
+
+    article.insertRecord();
+
+    // REVIEW: The following line of code redirects the user back to the home page after submitting the form.
+    window.location = '../';
+  }
+
+  articleView.initIndexPage = () => {
+    app.Article.all.forEach(a => $('#articles').append(a.toHtml()));
+    articleView.populateFilters();
+    articleView.handleCategoryFilter();
+    articleView.handleAuthorFilter();
+    articleView.handleMainNav();
+    articleView.setTeasers();
+    $('pre code').each((i, block) => hljs.highlightBlock(block));
+  };
+
+  articleView.initAdminPage = () => {
+  // REVIEW: We use .forEach() here because we are relying on the side-effects of the callback function: appending to the DOM. The callback is not required to return anything.
+    app.Article.numWordsByAuthor().forEach(stat => $('.author-stats').append(template(stat)));
```

Comment:
You get an error here because `template` is not defined; make sure to compile your template before you use it.